src/main/webapp/app/exam/manage/exam-status.component.ts

```diff
@@ -0,0 +1,178 @@
+        ExamConfigurationStep.REGISTER_STUDENTS,
+        ExamConfigurationStep.GENERATE_STUDENT_EXAMS,
+        ExamConfigurationStep.PREPARE_EXERCISE_START,
+    ];
+    configurationStepToFlagMap = new Map<ExamConfigurationStep, boolean>();
+    examChecklist: ExamChecklist;
+    numberOfGeneratedStudentExams: number;
+
+    examPreparationFinished = false;
+    examConductionState: ExamConductionState;
+    examReviewState: ExamReviewState;
+
+    readonly examConfigurationStepEnum = ExamConfigurationStep;
+    readonly examConductionStateEnum = ExamConductionState;
+    readonly examReviewStateEnum = ExamReviewState;
+    readonly round = round;
+    readonly Math = Math;
+
+    // Icons
+    faTimes = faTimes;
+    faTimesCircle = faTimesCircle;
+    faCheckCircle = faCheckCircle;
+    faArrowRight = faArrowRight;
+    faDotCircle = faDotCircle;
+    faCalendarCheck = faCalendarCheck;
+    faCalendarTimes = faCalendarTimes;
+
+    constructor(private examChecklistService: ExamChecklistService) {}
+
+    ngOnChanges(): void {
+        // Step 1: Exam preparation
+        // Step 1.1: Configure exercises
+        this.configurationStepToFlagMap.set(ExamConfigurationStep.CONFIGURE_EXERCISES, this.areAllExercisesConfigured());
+        // Step 1.2: Register students
+        this.configurationStepToFlagMap.set(ExamConfigurationStep.REGISTER_STUDENTS, this.examChecklistService.checkAtLeastOneRegisteredStudent(this.exam));
+        this.examChecklistService.getExamStatistics(this.exam).subscribe((examStats) => {
+            // Step 1.3: Generate student exams
+            this.configurationStepToFlagMap.set(ExamConfigurationStep.GENERATE_STUDENT_EXAMS, this.examChecklistService.checkAllExamsGenerated(this.exam, examStats));
+            // Step 1.4: Prepare exercise start
+            const allExamsGenerated = this.configurationStepToFlagMap.get(ExamConfigurationStep.GENERATE_STUDENT_EXAMS)!;
+            this.configurationStepToFlagMap.set(ExamConfigurationStep.PREPARE_EXERCISE_START, !!examStats.allExamExercisesAllStudentsPrepared && allExamsGenerated);
+            this.examPreparationFinished = this.isExamPreparationFinished();
+            this.examChecklist = examStats;
+            this.numberOfGeneratedStudentExams = this.examChecklist.numberOfGeneratedStudentExams ?? 0;
+        });
+
+        // Step 2: Exam conduction
+        this.setConductionState();
+
+        // Step 3: Exam correction
+        this.setReviewState();
+    }
+
+    /**
+     * Auxiliary method that determines if all configuration steps for the exam exercises are finished. There are 5 requirements that have to be fulfilled:
+     * 1. The exam has at least one exercise group
+     * 2. The number of exam exercises equals number of exercise groups
+     * 3. Each exercise group contains at least one exercise
+     * 4. Within each exercise group, exercises have same maximum points
+     * 5. Maximum points of exam are reachable
```

Comment:
```suggestion
     * Auxiliary method that determines if all configuration steps for the exam exercises are finished.
```
I don’t think the additional list is necessary and neither are the inline comments inside the method, as the method names have been chosen nicely to already explain what is checked.